src/main/kotlin/kweb/Kweb.kt

```diff
@@ -295,147 +315,137 @@ class Kweb private constructor(
+                            handleError(message.error, remoteClientState)
+                        } else {
+                            when {
+                                message.callback != null -> {
+                                    val (resultId, result) = message.callback
+                                    val resultHandler = remoteClientState.handlers[resultId]
+                                            ?: error("No data handler for $resultId for client ${remoteClientState.id}")
+                                    resultHandler(result ?: "")
+                                }
+                                message.historyStateChange != null -> {
 
-            try {
-                for (frame in incoming) {
-                    try {
-                        logger.debug { "Message received from client" }
-
-                        if (frame is Text) {
-                            val message = gson.fromJson<Client2ServerMessage>(frame.readText())
-                            logger.debug { "Message received: $message" }
-                            if (message.error != null) {
-                                handleError(message.error, remoteClientState)
-                            } else {
-                                when {
-                                    message.callback != null -> {
-                                        val (resultId, result) = message.callback
-                                        val resultHandler = remoteClientState.handlers[resultId]
-                                                ?: error("No data handler for $resultId for client ${remoteClientState.id}")
-                                        resultHandler(result ?: "")
-                                    }
-                                    message.historyStateChange != null -> {
-
-                                    }
                                 }
                             }
                         }
-                    } catch (e: Exception) {
-                        logger.error("Exception while receiving websocket message", e)
                     }
+                } catch (e: Exception) {
+                    logger.error("Exception while receiving websocket message", e)
                 }
-            } finally {
-                logger.info("WS session disconnected for client id: ${remoteClientState.id}")
-                remoteClientState.clientConnection = Caching()
             }
+        } finally {
+            logger.info("WS session disconnected for client id: ${remoteClientState.id}")
+            remoteClientState.clientConnection = Caching()
         }
     }
 
-    private fun Routing.listenForHTTPConnection(htmlDocumentTemplate: Document) {
-        get("/{visitedUrl...}") {
-            val htmlDocument = htmlDocumentTemplate.clone()
+    private suspend fun ApplicationCall.listenForHTTPConnection(htmlDocument: Document, buildPage: WebBrowser.() -> Unit) {
```

Comment:
Also very, very little changes here. Basically 
- I pass the buildPage as an arg instead of using the field access
- Extracted the `get(...)` call for visibility
- Changed receiver to ApplicationCall, since that's what we need
- formatting to the new world